projects/knora-ui/src/lib/viewer/values/link-value/link-value.component.ts

```diff
@@ -0,0 +1,175 @@
+import {Component, Inject, Input, OnChanges, OnDestroy, OnInit, SimpleChanges} from '@angular/core';
+import {BaseValueComponent} from '../base-value.component';
+import {CreateLinkValue, ReadLinkValue, ReadResource, UpdateLinkValue, KnoraApiConnection, IResourceClassAndPropertyDefinitions} from '@knora/api';
+import {Subscription} from 'rxjs';
+import {AbstractControl, FormBuilder, FormControl, FormGroup, ValidatorFn} from '@angular/forms';
+import {KnoraApiConnectionToken} from '../../../core';
+@Component({
+  selector: 'kui-link-value',
+  templateUrl: './link-value.component.html',
+  styleUrls: ['./link-value.component.scss']
+})
+export class LinkValueComponent extends BaseValueComponent implements OnInit, OnChanges, OnDestroy {
+  @Input() displayValue?: ReadLinkValue;
+  // @Input() propertyDesc?: IResourceClassAndPropertyDefinitions;
+  resources: ReadResource[];
+  restrictToResourceClass: string;
+  valueFormControl: FormControl;
+  commentFormControl: FormControl;
+  form: FormGroup;
+
+  valueChangesSubscription: Subscription;
+  // label cannot contain logical operations of lucene index
+  customValidators = [];
+
+  constructor(@Inject(FormBuilder) private fb: FormBuilder, @Inject(KnoraApiConnectionToken) private knoraApiConnection: KnoraApiConnection) {
+    super();
+  }
+
+  /**
+   * Displays a selected resource using its label.
+   *
+   * @param resource the resource to be displayed (or no selection yet).
+   * @returns
+   */
+  displayResource(resource: ReadResource | null): string {
+
+    // null is the initial value (no selection yet)
+    if (resource !== null) {
+      return resource.label;
+    }
+  }
+
+  /**
+   * Search for resources whose labels contain the given search term, restricting to to the given properties object constraint.
+   * this is to be used for update and new linked resources
+   *
+   * @param label to be searched
+   */
+  searchByLabel(searchTerm: string): ReadResource[] {
+    // todo: change the statement below once the function is added to knora-api.js.lib to get link property defintion
+    this.restrictToResourceClass = 'http://0.0.0.0:3333/ontology/0001/anything/v2#Thing';
+    // at least 3 characters are required
+    if (searchTerm.length >= 3) {
+
+      this.knoraApiConnection.v2.search.doSearchByLabel(searchTerm, 0, {limitToResourceClass: this.restrictToResourceClass} ).subscribe(
+        (response: ReadResource[]) => {
+          this.resources = response;
+        });
+    } else {
+      // clear selection
+      this.resources = undefined;
+
+    }
+    return this.resources;
+  }
+
+  // show the label of the linked resource
+  getInitValue(): ReadResource | null {
+    if (this.displayValue !== undefined) {
+
+      // console.log(this.displayValue);
+      // console.log(this.propertyDesc.properties[this.displayValue.property]);
+      return this.displayValue.linkedResource;
```

Comment:
Actually this member is optional. In this context it is save to assume that the linked resource is present. However, when dealing with Gravsearch results it is not (a link value might only contain the Iri of the linked resource but no embedded object to prevent circular references).